lib/ethers/transaction.ex

```diff
@@ -109,6 +109,50 @@ defmodule Ethers.Transaction do
     |> prepend_type_envelope(type)
   end
 
+  # Assuming the input is always a single string, the signed tx
+  # We could just leave the (most likely present) 0x on and just pass it around,
+  # i don't think hex_decode cares, but it makes inspecting the possible prexfix easier to remove it - discuss
+  def decode("0x" <> signed_tx), do: decode(signed_tx) 
+  def decode(signed_tx), do: decode(decode_tx_and_type(signed_tx))
+  def decode(:eip1559, decoded) do
+    # TODO we could inspect the length and throw if not 12..
+    # [chain_id, nonce, max_priority_fee, max_fee, gas_limit, to, value, data, access_list, y, r, s]
+    vector = ExRLP.decode(decoded)
+
+    # no recover the public key (from), we first need to reconstruct the unsigned 'digest'
+    # NOTE: after the digest is assembled we also prepend the type envelope
+    tx_hash =
+      Enum.take(vector, 9)
+      |> ExRLP.encode()
+      |> prepend_type_envelope(:eip1559)
+      |> ExKeccak.hash_256()
+
+    [y, r, s] = Enum.take(vector, -3)
+
+    {:ok, public_key} = ExSecp256k1.recover(tx_hash, r, s, :binary.decode_unsigned(y))
+
+    {:ok, struct!(__MODULE__, %{
```

Comment:
@alisinabh i'm assuming we want to return a transaction struct from `decode/1`

Reply:
Hey, Sorry I somehow missed this.

Yes it should.

Do you still want to work on this?